src/sync/outgoingSyncUtils.js

```diff
@@ -51,16 +53,37 @@ export function generateSyncJson(database, settings, syncOutRecord) {
   } else {
     // Get the record the syncOutRecord refers to from the database
     const recordResults = database.objects(recordType).filtered('id == $0', recordId);
-    if (!recordResults || recordResults.length === 0) { // No such record
+    if (!recordResults || recordResults.length === 0) {
+      // No such record
       throw new Error(`${recordType} with id = ${recordId} missing`);
-    } else if (recordResults.length > 1) { // Duplicate records
+    } else if (recordResults.length > 1) {
+      // Duplicate records
       throw new Error(`Multiple ${recordType} records with id = ${recordId}`);
     }
     const record = recordResults[0];
 
     // Generate the appropriate data for the sync object to carry, representing the
     // record in its upstream form
-    syncData = generateSyncData(settings, recordType, record);
+    try {
+      syncData = generateSyncData(settings, recordType, record);
+    } catch (error) {
+      // There was an error with data, often a null object
+      const siteName = settings.get(SYNC_SITE_NAME);
+      const syncUrl = settings.get(SYNC_URL);
+      const originalMessage = error.message;
+      // Change error message to be helpful in bugsnag
+      error.message =
+        `SYNC OUT ERROR. siteName: ${siteName}, serverUrl: ${syncUrl}, ` +
```

Comment:
should this be a json, rather then a string ?

Reply:
Worked correctly so far. Error message is meant to be a string from how I interpret mdn docs.

Reply:
I was thinking from neatness point of view. I guess it's neat enough as it is (interesting to see string concatenation + template literal, I guess eslint allows multi line concatenation).

Reply:
It's mostly just against concatenating string literals where a template string would be better.